frontend/src/Onboarding/LoginScreen.tsx

```diff
@@ -0,0 +1,320 @@
+import Cookies from "js-cookie";
+import { AppState } from "../state/reducers/state";
+import { SecondaryLinkButton } from "../components/common/LinkButtons";
+
+const Wrapper = styled.div`
+  display: flex;
+  flex: 1;
+  flex-direction: column;
+  align-items: center;
+  height: 100vh;
+`;
+
+const Title = styled.div`
+  margin-top: 96px;
+  font-style: normal;
+  font-weight: bold;
+  font-size: 24px;
+  line-height: 28px;
+  color: #000000;
+`;
+
+const Subtitle = styled.div`
+  margin-top: 8px;
+  margin-bottom: 4px;
+  font-family: Roboto;
+  font-style: normal;
+  font-weight: normal;
+  font-size: 14px;
+  line-height: 16px;
+  width: 326px;
+  text-align: center;
+`;
+
+const Box = styled.div`
+  border: 1px solid white;
+  width: 500px;
+  align-items: center;
+  display: flex;
+  flex-direction: column;
+`;
+
+const ButtonContainer = styled.div`
+  margin-top: 1em;
+  display: grid;
+  grid-template-columns: auto auto;
+  gap: 0em 1em;
+`;
+
+interface ReduxStoreLoginScreenProps {
+  setAcademicYear: (academicYear: number) => void;
+  setGraduationYear: (graduationYear: number) => void;
+  setUserCoopCycle: (coopCycle: string) => void;
+  setSchedules: (schedules: NamedSchedule[]) => void;
+  setPlanStr: (planStr: string) => void;
+  setStudent: (student: IUserData) => void;
+}
+interface ReduxStoreSignupScreenProps {
+  majors: Major[];
+  setEmail: (email: string) => void;
+}
+
+type Props = ReduxStoreLoginScreenProps &
+  ReduxStoreSignupScreenProps &
+  RouteComponentProps<{}>;
+
+enum ErrorType {
+  INVALID_CREDENTIALS,
+  OTHER,
+}
+
+interface LoginScreenState {
+  emailStr: string;
+  passwordStr: string;
+  beenEditedEmail: boolean;
+  beenEditedPassword: boolean;
+  validEmail: boolean;
+  error?: ErrorType;
+}
+
+class LoginScreenComponent extends React.Component<Props, LoginScreenState> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      emailStr: "",
+      passwordStr: "",
+      beenEditedEmail: false,
+      beenEditedPassword: false,
+      validEmail: true,
+      error: undefined,
+    };
+  }
+
+  /**
+   * All of the different functions that modify the stored TextField values as they are changed.
+   */
+  onChangeEmail(e: React.ChangeEvent<HTMLInputElement>) {
+    this.setState({
+      emailStr: e.target.value,
+      beenEditedEmail: true,
+    });
+  }
+
+  onChangePassword(e: React.ChangeEvent<HTMLInputElement>) {
+    this.setState({
+      passwordStr: e.target.value,
+      beenEditedPassword: true,
+    });
+  }
+
+  /**
+   * Validates user input, then sends a log in request to the backend using the input data.
+   * Checks response for error messages. If the log in succeeds, dispatch actions to set
+   * user attributes obtained from the response object, then redirects user to /home.
+   */
+  async submit() {
+    // Regex to determine if email string is a valid address
+    const validEmail: boolean = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(
```

Comment:
kinda talked about this during the meeting too. I think we should use some packages to do the validation and onSubmit (like `formik` you suggested and `yup`). If it's too much of a change in this PR, it can be done on a separate one.